cli/dts/lib.deno.unstable.d.ts

```diff
@@ -1196,6 +1196,21 @@ declare namespace Deno {
     bytesSentData: number;
     bytesReceived: number;
   }
+
+  export interface HttpEvent {
+    readonly request: Request;
+    respondWith(r: Response | Promise<Response>): void;
+  }
+
+  /** **UNSTABLE**: new API, yet to be vetted.
+   *
+   * Parse HTTP requests from the given connection
+   *
+   * ```ts
+   * const requests = await Deno.startHttp(conn);
+   * ```
+   */
+  export function startHttp(conn: Conn): AsyncIterableIterator<HttpEvent>;
```

Comment:
We should downgrade this return type to `AsyncIterable<HttpEvent>`, and now or in the future upgrade that to `HttpConn extends AsyncIterable<HttpEvent>` which exposes `.rid` and `.close()` if we want those things.

Returning or extending `AsyncIterableIterator<_>` (and overloading `.return()` for closing) isn't the way to go. I pushed against that for `Listener` early on, but unfortunately `watchFs()` slipped through and demonstrated this. I've been meaning to fix that for 2.0.

Reply:
> Returning or extending AsyncIterableIterator<_> (and overloading .return() for closing) isn't the way to go

What's your reasoning behind this?

Reply:
For starters `.return()` is optional on `AsyncIterableIterator`: https://github.com/denoland/deno/blob/main/cli/tests/unit/fs_events_test.ts#L74. It doesn't align with the default implementation of `.return()` on generators, which is supposed to break any pending loops without error. I don't think this is supposed to be manually implemented.

The iterator protocol methods like `.next()` aren't a nice thing to directly expose on structures with other things, it would be better to have just `[Symbol.asyncIterator]()` and maybe add an `accept()` method to await one request like with `Listener`. However we don't actually need to adjust the implementation from what it is now since it's probably more efficient -- that would also match `Listener`.

Reply:
Ok. I have changed this around so that there is no `return` method and that the requests iterator must be closed manually. Is this what you had in mind?

Reply:
> and that the requests iterator must be closed manually

The resource should still be closed automatically if `connectionClosed` is true, if I seemed to have suggested otherwise.